Sources/HTTPTypes/HTTPFieldName.swift

```diff
@@ -77,6 +77,34 @@ extension HTTPField.Name: CustomPlaygroundDisplayConvertible {
     }
 }
 
+extension HTTPField.Name: Codable {
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(self.rawName)
+    }
+
+    private enum DecodingError: Error {
+        case invalidString(String)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let nameString = try container.decode(String.self)
+        if nameString.hasPrefix(":") {
+            guard nameString.lowercased() == nameString,
+                  HTTPField.isValidToken(nameString.dropFirst()) else {
+                throw DecodingError.invalidString(nameString)
+            }
+            self.init(rawName: nameString, canonicalName: nameString)
```

Comment:
Is there a reason we use different initializers in these two code paths?

Reply:
It's not possible to construct a pseudo header field name using the default constructor. `:method` is not a valid HTTP token. We deliberately made it so that pseudo header fields are built-in and only constructible by us.